numbat/examples/inspect.rs

```diff
@@ -86,25 +79,137 @@ fn inspect_functions_in_module(ctx: &Context, module: String) {
         println!("{signature}");
         println!("```");
         println!();
+
+        if !examples.is_empty() {
+            println!("<details>");
+            println!("<summary>Examples</summary>");
+            println!();
+
+            for (example_code, example_description) in examples {
+                let mut example_ctx = prelude_ctx.clone();
+                let extra_import = if !example_ctx
+                    .resolver()
+                    .imported_modules
+                    .contains(&module_path)
+                {
+                    format!("use {}\n", module)
+                } else {
+                    "".into()
+                };
+                let _result = example_ctx
+                    .interpret(&extra_import, CodeSource::Internal)
+                    .unwrap();
+
+                if let Ok((statements, results)) =
+                    example_ctx.interpret(&example_code, CodeSource::Internal)
+                {
+                    //Format the example input
+                    let example_input = format!(">>> {}", example_code);
+
+                    //Encode the example url
+                    let url_code = extra_import + &example_code;
+                    let example_url = format!(
+                        "https://numbat.dev/?q={}",
+                        percent_encoding::utf8_percent_encode(
+                            &url_code,
+                            percent_encoding::NON_ALPHANUMERIC
+                        )
+                    );
+
+                    //Assemble the example output
+                    let mut example_output = String::new();
+                    example_output += "\n";
+
+                    for statement in &statements {
+                        example_output += &plain_text_format(&statement.pretty_print(), true);
+                        example_output += "\n\n";
+                    }
```

Comment:
Do you think we could skip pretty printing, or do you see any problems with this for some of the examples? For most of the examples, I think it only adds visual noise:

![image](https://github.com/user-attachments/assets/ac0beba5-5014-4934-ab6e-bc8552192ced)

Reply:
I think pretty printing can be helpful for some examples, e.g to see how functions are typed, but for most examples it doesn't add much value so I am fine to remove it to make the examples more concise.

Reply:
> I am fine to remove it to make the examples more concise.

That would be great, thanks.